src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/get_options_list_control_factory.tsx

```diff
@@ -292,6 +292,7 @@ export const getOptionsListControlFactory = (): DataControlFactory<
             hideExclude: 'skip',
             hideExists: 'skip',
             hideSort: 'skip',
+            disableMultiValueEmptySelection: 'skip',
```

Comment:
Is there any reason this needs to be _state_? If I'm understanding correctly, you'd want to disallow deselecting all for _all_ ESQL options lists - maybe it can be a prop passed in instead?

Reply:
We generally like to avoid putting things in the state unless they should be saved with the Dashboard, _and_ change over the lifetime of the control. And this ethos is more important to us generally than avoiding prop drilling, as if it's really egregious other methods can be used to avoid it.

Keys like `hideSort` & `hideExclude` are in the state _unfortunately_, because they need to be passed in as configuration items in solution usages of the control group, and the embeddable system these controls are based on provides no non-state channel for that kind of communication.